src/log.ts

```diff
@@ -176,8 +176,12 @@ export namespace message {
       channel === 'x' ? 'width' : 'height'} is provided.`;
   }
 
-  export function scaleTypeNotWorkWithChannel(channel: Channel, scaleType: ScaleType, newScaleType: ScaleType) {
-    return `Channel ${channel} does not work with ${scaleType} scale. We are using ${newScaleType} scale instead.`;
+  export function cannotOverrideBinScaleType(channel: Channel) {
+    return `Cannot override scale type for binned channel ${channel}.`;
+  }
+
+  export function scaleTypeNotWorkWithChannel(channel: Channel, scaleType: ScaleType) {
```

Comment:
>  we had to compute the right scale twice

That's not true.  We compute it *once* (but have two separate lines that call it, which is totally okay.)  

>  it requires some expertise to change the scale type anyway.

I disagree.  It lets users know what we automatically do for them, which is important (making the system transparent.)

> You can just look at the vega output.

This option is not always available for users (e.g., in Altair).